<!-- This is an example launch file for how a ROS application could use a node -->
<launch>
   
	<arg name="use_pattern" default="true"/>
	<arg name="test" default="false"/>
	<arg name="input_test" default="Hey"/>
	<arg name="id_test" default="default"/>
	<arg name="test_service" default=""/>
	<arg name="tts_input_text" default="My name is HARMONI"/>
	<arg name="dialogue_input_text" default="Hey"/>
	<arg name="display_input" default="{'component_id':'container_1', 'set_content': 'null'}"/>
	<arg name="face_input" default="[{'start': 0.075, 'time': 2,'type': 'action', 'id': 'QT/point_front'}, {'start': 0.075,'time': 2, 'type': 'viseme', 'id': 'POSTALVEOLAR'},{'start': 0.006, 'time': 2,  'type': 'action', 'id': 'happy_face'}]"/>
	<arg name="wav_file" default="/home/micol/catkin_ws/src/HARMONI/harmoni_actuators/harmoni_tts/temp_data/tts.wav"/>
  	<group if="$(arg use_pattern)">
			<rosparam file="$(find harmoni_pattern)/config/configuration.yaml"/>
			<param name="test_service" value="$(arg test_service)"/>
			<param name="wav_file" value="$(arg wav_file)"/>
			<param name="face_input" value="$(arg face_input)"/>
			<param name="display_input" value="$(arg display_input)"/>
			<param name="tts_input_text" value="$(arg tts_input_text)"/>
			<param name="dialogue_input_text" value="$(arg dialogue_input_text)"/>
			<param name="test" value="$(arg test)"/>
			<param name="input_test" value="$(arg input_test)"/>
			<param name="id_test" value="$(arg id_test)"/>
			<node pkg="harmoni_pattern" type="dialoging_pattern.py" name="harmoni_pattern" output="screen"  args="$(arg test) $(arg input_test) $(arg id_test)"/>
  	</group>


</launch>
